### ISSUE TITLE
Decide on data storage based on needs.

### ISSUE THREAD

**Issue by JJ** [Stance4Health-Dev/docs#4]
* Check out Neo4J or other graph database
* Check out Bio4J or its underlying database (Vulcan, I think)

**Comment by thebooort** [comment]
@JJ Suggested today the usage of Elastic https://www.elastic.co/
(is that the right link?)

**Comment by Gecofer** [comment]
> @JJ Suggested today the usage of Elastic https://www.elastic.co/
> (is that the right link?)

This sample application demonstrates:
- Indexing recipes using the Java Client API and
- Using the Java Client API to searching for recipes by keywords

https://github.com/elastic/examples/tree/master/Search/recipe_search_java

**Comment by JJ** [comment]
Good catch, also. Trying to find the way it's stored, but it's not easy.

**Comment by JJ** [comment]
Ah, it's a JSON:
https://github.com/elastic/examples/blob/master/Search/recipe_search_java/data/banana-oatmeal-cookie.json

**Comment by jimcase** [comment]
I found this interesting article: https://thenewstack.io/tinkerpop-growing-graph-database-popularity/
In Summary:
"Tinkerpop is a graph computing framework for tying together relational and graph databases in one unified model, which just became a top-level project with the Apache Software Foundation."
http://tinkerpop.apache.org/
Tinkerpop  use  Gremlin, "a graph traversal machine and language, which enables users to write complex queries called traversals that can execute either as real-time transactional (OLTP) queries, batch analytic graph (OLAP) queries, or a hybrid of the two"

It's open source
Amazon use it for its order fulfillment graph

Documentation: http://tinkerpop.apache.org/docs/current/reference/
Tutorial: https://kelvinlawrence.net/book/Gremlin-Graph-Guide.html
Gremlin-Python for Apache TinkerPop: https://pypi.org/project/gremlinpython/

With Tinkerpop  3.4.0 API we can use the next database tools: http://tinkerpop.apache.org/javadocs/current/full/
Some examples:
- Neo4j
- Hadoop
- Elastic (https://github.com/rmagen/elastic-gremlin)
- Spark
- Apache Giraph

**Comment by jimcase** [comment]
Github repo: https://github.com/apache/tinkerpop
Last release 3.4.0 in 04/01/209, so currently maintained and updated

**Comment by jimcase** [comment]
Azure Cosmos DB can be used with Gremlin API to build graphs storage.

Introduction to Azure Cosmos DB: Gremlin API
https://docs.microsoft.com/en-us/azure/cosmos-db/graph-introduction

Being more specific to our domain:
Quickstart: Create a graph database in Azure Cosmos DB using Python and the Azure portal
https://docs.microsoft.com/en-us/azure/cosmos-db/create-graph-python
